templates/contribute/pr-review.md

```diff
@@ -0,0 +1,715 @@
+code duplication; this leads to the canonical questions by new users: "is mathlib really
+missing vector spaces and group homomorphisms?", and its answer: "no, these are `Module`
+and `MonoidHom`, respectively."
+
+As a result, it is not uncommon for new contributors (or even experienced ones!) to 
+create a PR for a result that already exists, sometimes verbatim, and other times in
+greater generality.
+
+The `apply?` and `exact?` tactics can help answer this sometimes. If you suspect a 
+result already exists, just copy it to a new file with `import Mathlib` and try `exact?`.
+
+#### Do new `import`s exacerbate import creep?
+
+Maintaining the organization of the mathlib import hierarchy is an important task, 
+but it can easily get out of hand without careful review. Generally, the problem occurs
+in the following manner.
+
+A contributor thinks: "I would like to add `my_theorem` and it's all about `Z`, so I'll
+add it to `X.Y.Z`." Upon trying to add the theorem there, the contributor realizes: "oh,
+I don't have access to `helper_lemma`, I need to `import A.B.C`." During review, the 
+reviewer is focused on other things, and the PR is merged with this import change.
+This is the story of how, at one time, `Analysis.NormedSpace.Star.Basic` imported 
+`Analysis.NormedSpace.OperatorNorm`!
+
+As another example, in [#6239](https://github.com/leanprover-community/mathlib4/pull/6239),
+the contributor had added the import `Data.IsROrC.Basic` to
+`LinearAlgebra.Matrix.DotProduct`. This is probably a hard thing for a new 
+contributor to recognize, because it sometimes requires decent familiarity with the
+way the library is organized.
+
+Of course, reviewers should do their best to catch the most egregious examples
+(e.g., importing `Analysis` files into `Algebra` files is generally rather suspect),
+but asking the question is always warranted. It can often mean that the results belong
+elsewhere, the file should be split along a natural boundary, or the new results should
+go in a new a file.
+
+#### Should a file be split into multiple pieces?
+
+There are essentially three reasons to split a file:
+
+1. It is simply too long to be nice to use.
+   A good rule of thumb here is that it exceeds 1000 lines.
+2. The file is sectioned into multiple pieces that are only loosely related.
+3. To avoid import creep from the introduction of new results that are closely
+   related to existing ones.
+
+Let's say some results, about 500 lines worth, are added to an existing file which 
+already contains 700 lines, and suppose further that the new material is closely 
+related to some of the existing material in the file. A reviewer should be on the 
+lookout for a natural boundaries where the file can be split into coherent pieces.
+
+### Improvements
+
+#### Splitting into supporting lemmas or definitions (especially for long proofs)?
+
+I don't have an example of this on hand.
```

Comment:
After some semi-random poking around, I also couldn't find an example of this. Maybe a few sentences about how standalone long proofs can sometimes be a "code smell" and a sign for the reivewer to think a bit about possible refactors?